### ISSUE TITLE
Information about curves on belts/pipes ?

### ISSUE THREAD

**Issue by ReicoC** [demipixel/factorio-blueprint#9]
Hey i am trying to write something similar (to create an image) and wanted to ask how you get information about if a pipe or belt is a curve.
Currently we have the orientations up/down/left/right. We wonder how we get the information if it is a curved belt piece and in which direction it is facing?

Do you have an idea or maybe already implemented that?

Greetings

**Comment by demipixel** [comment]
Nope, factorio does it by checking the orientation and looking at surrounding entities. Assuming you have a belt facing right, the bool test for "if the belt should be turned" would be:

`(NOT belt behind) AND (belt below XOR belt above)`

If you're writing your own library, you could cache this data and update surrounding entities when a new entity is added/removed, but that's up to you.
